packages/x-charts/src/LineChart/AreaElement.tsx

```diff
@@ -12,6 +14,8 @@ export interface AreaElementClasses {
 export interface AreaElementOwnerState {
   id: string;
   color: string;
+  isNotHighlighted: boolean;
+  isHighlighted: boolean;
```

Comment:
We can technically go with `true | false | undefined` But it's something that could be used a lot, and it feels more natural to write customization as follow:

```
{
  color: blue,
  opacity: 0.8,
  ...(ownerState.isHighlighted && {
    color: darkblue,	
    opacity: 1,
  }),
  ...(ownerState.isNotHighlighted && {
    color: darkblue,	
    opacity: 1,
  }),
}
```

Otherwise, it will implies to get some `ownerState.isHighlighted === false` which could be confusing especially for new commers.

But it's a tradeoff tat can be debated

Reply:
Indeed, being explicit and more terse is often preferred over too simplified and cumbersome approach. 👌